ansible/environments/distributed/group_vars/all

```diff
@@ -70,3 +70,4 @@ instances:
   - name: db
     num_instances: 1
     flavor:
+    volume:
```

Comment:
See comments regarding disk sizes below. The same works for instance flavors here.

Instead of describing specific host groups in the `group_vars/all` file which is applied to all hosts, you could provide individual group_vars files for each host group which would make this list obsolete.

e.g.

group_vars/edge
```
flavor: small
```

group_vars/controllers
```
flavor: medium
```

etc.

the `num_instances` property is redundant because the number of hosts is already defined in the `hosts` inventory file per each group.

for provisioning you would have a playbook that simply targets all hosts and uses their flavor to provision the right size:

```
- hosts: all
  tasks:
    - create volume
    - provision host
    - attach volume
    ....
```

each task could simply access variables as defined in the proper group_vars/xxx file without the need to `include_vars` and weird `set_facts` for the instance name. The instance name is already defined in your `hosts` inventory file. It is `inventory_hostname`. Why define it a second time?